actionmailer/lib/action_mailer/delivery_job.rb

```diff
@@ -12,9 +12,16 @@ class DeliveryJob < ActiveJob::Base # :nodoc:
 
     rescue_from StandardError, with: :handle_exception_with_mailer_class
 
-    def perform(mailer, mail_method, delivery_method, params, *args) #:nodoc:
-      mailer_class = params ? mailer.constantize.with(params) : mailer.constantize
-      mailer_class.public_send(mail_method, *args).send(delivery_method)
+    before_perform do
+      ActiveSupport::Deprecation.warn <<~MSG.squish
+        Sending mail with DeliveryJob and Parameterized::DeliveryJob
```

Comment:
Since `Parameterized::DeliveryJob` isn't released yet can we instead deprecate just `DeliveryJob` and then recommend the new one? If you need the `params` key then you should use the new one.

Reply:
I removed `Parameterized::DeliveryJob` from the warning, but the warning will still happen for parameterized delivery jobs because its a subclass. I assume this is good enough?

Reply:
I think it is released in all versions since [`5.1.0`](https://github.com/rails/rails/blob/v5.1.0/actionmailer/lib/action_mailer/parameterized.rb#L146)

Reply:
Ah I see I was thinking of this PR https://github.com/rails/rails/commit/251b9d4bd62bfc076969bd0751fa4cc6e95a4f33 which is only in master.